### PR TITLE
Scoring

### PR DESCRIPTION
**Pull request by SimplemathicDev**
Testing testing...

Added a scoring system with 5 possible rules (modify the _scoringRule variable):
- Points only for time
- Points only for level-up
- Points for both
- Points only for level-up, decreased over time
- Points only for time, based on level

Known issue: score accumulates during countdown

### CODE REVIEW

File: weave/Scripts/Score.cs [hunk 1]
```diff
@@ -0,0 +1,72 @@
+using Godot;
+using GodotSharper.AutoGetNode;
+using System;
+using weave.Utils;
+
+namespace weave;
+
+public partial class Score : CanvasLayer
+{
+	private enum ScoringRule
```
Comment (theowiik) [inline weave/Scripts/Score.cs:10]: Indented with tabs, could you format it with CSharpier? (or just 4 spaces)

https://github.com/weave-game/weave#formatting

File: weave/Scripts/Score.cs [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+				break;
+			default:
+				break;
+		}
+
+		_timeSinceLevelStart += delta;
+		_scoreLabel.Text = ((int)_score).ToString();
+	}
+
+	public void OnLevelUp(int currentLevel)
```
Comment (theowiik) [inline weave/Scripts/Score.cs:53]: Is this used?

File: weave/Scripts/Utils/ScoringConstants.cs [hunk 3]
```diff
@@ -0,0 +1,9 @@
+namespace weave.Utils;
+
+public static class ScoringConstants
+{
+    public const int PointsForSeconds = 25;
+    public const int PointsForLevelUp = 500;
```
Comment (theowiik) [inline weave/Scripts/Utils/ScoringConstants.cs:6]: We should probably have these as floats as we will lose whole digits with delta multiplication

**Review by theowiik — CHANGES_REQUESTED** [review]

File: weave/Scripts/Score.cs (hunk 2 above)
Comment (SimplemathicDev) [reply weave/Scripts/Score.cs:53]: Thanks for pointing that out, forgot to call it from Main.cs

File: weave/Scripts/Utils/ScoringConstants.cs (hunk 3 above)
Comment (SimplemathicDev) [reply weave/Scripts/Utils/ScoringConstants.cs:6]: Isn't delta double? After the multiplication with delta, the result is stored in _score also as double, so I don't understand why any accuracy would be lost.

Comment (theowiik) [reply weave/Scripts/Utils/ScoringConstants.cs]: You're right! My bad :+1:

**Review by theowiik — APPROVED** [review]

**Pull request merged**